fab/target_distributions/many_well.py

```diff
@@ -108,8 +108,9 @@ def performance_metrics(self, samples: torch.Tensor, log_w: torch.Tensor,
             log_w = log_w[:n_vals_per_split*n_runs]
             log_w = torch.stack(log_w.split(n_runs), dim=-1)
             # Check accuracy in estimating normalisation constant.
-            Z_estimate = torch.exp(torch.logsumexp(log_w, axis=-1) - np.log(log_w.shape[-1]))
-            MSE_Z_estimate = torch.mean(torch.abs((Z_estimate - self.Z) / self.Z))
+            log_Z_estimate = torch.logsumexp(log_w, axis=-1) - np.log(log_w.shape[-1])
```

Comment:
The previous way I calculated this was only stable for float 64 (broke for float32)